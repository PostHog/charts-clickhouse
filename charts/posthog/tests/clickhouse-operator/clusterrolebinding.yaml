suite: ClickHouse cluster role binding definition
templates:
  - templates/clickhouse-operator/clusterrolebinding.yaml

tests:
  - it: should be empty if clickhouseOperator.enabled is set to false
    set:
      clickhouseOperator.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: the manifest should match the snapshot when using default values
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: ClusterRoleBinding/clickhouse-operator-posthog metadata.namespace override via clickhouseOperator.namespace should work
    set:
      clickhouseOperator.namespace: "custom-namespace"
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.namespace
          value: custom-namespace

  - it: ClusterRoleBinding/clickhouse-operator-posthog metadata.namespace override via Release.Namespace should work
    release:
      namespace: custom-namespace-from-release
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.namespace
          value: custom-namespace-from-release

  - it: ClusterRoleBinding/clickhouse-operator-posthog subjects[0].namespace override via clickhouseOperator.namespace should work
    set:
      clickhouseOperator.namespace: "custom-namespace"
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: subjects[0].namespace
          value: custom-namespace

  - it: ClusterRoleBinding/clickhouse-operator-posthog subjects[0].namespace override via Release.Namespace should work
    release:
      namespace: custom-namespace-from-release
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: subjects[0].namespace
          value: custom-namespace-from-release
