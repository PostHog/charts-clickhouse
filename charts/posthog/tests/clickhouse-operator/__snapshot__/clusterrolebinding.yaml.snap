the manifest should match the snapshot when using default values:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      labels:
        clickhouse.altinity.com/chop: 0.19.0
      name: clickhouse-operator-NAMESPACE
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: clickhouse-operator-NAMESPACE
    subjects:
    - kind: ServiceAccount
      name: clickhouse-operator
      namespace: NAMESPACE
